### PR TITLE
Pad bandwidth bound split k kernels on a100

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125650

For 
```
import torch
import triton

dtype = torch.bfloat16

t1 = torch.empty([2, 20569856], dtype=dtype, device="cuda")
t2 = torch.empty([20569856, 13], dtype=dtype, device="cuda")

@torch.compile()
def benchmark(t1, t2):
    return torch.ops.aten.mm(t1, t2)

print(triton.testing.do_bench(lambda: benchmark(t1, t2)))
```

Improves perf from 449ms  -> 1.2578779458999634ms

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/fx_passes/pad_mm.py [hunk 1]
```diff
@@ -184,6 +184,15 @@ def is_mm_compute_bound(M: int, K: int, N: int, dtype: torch.dtype) -> bool:
         return False
     arithmetic_intensity = (M * N * K) / denominator
 
+    # we have experienced some large perf hits in this case, even in bandwidth bound regimes
+    if (
+        dtype is torch.bfloat16
+        and K > M
+        and K > N
+        and torch.cuda.get_device_capability() < (9, 0)
+    ):  # doenst repro on h100s:
```
Comment (bertmaher) [inline torch/_inductor/fx_passes/pad_mm.py:193]: minor typo `doenst -> doesn't`

File: torch/_inductor/fx_passes/pad_mm.py [hunk 2]
```diff
@@ -184,6 +184,15 @@ def is_mm_compute_bound(M: int, K: int, N: int, dtype: torch.dtype) -> bool:
         return False
     arithmetic_intensity = (M * N * K) / denominator
 
+    # we have experienced some large perf hits in this case, even in bandwidth bound regimes
+    if (
+        dtype is torch.bfloat16
+        and K > M
+        and K > N
```
Comment (bertmaher) [inline torch/_inductor/fx_passes/pad_mm.py:191]: This is pretty permissive, does that seem ok?  Any downsides?

**Review by bertmaher — APPROVED** [review]

File: torch/_inductor/fx_passes/pad_mm.py (hunk 2 above)
Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:191]: Happy to tighten it. The main reason we skipped bandwidth bound mms was bc of kv cache - you would have a bunch of mms that were all bandwidth bound of slightly different shape and it really increased compile time. Not an issue here..

We've seen pretty large perf issues with bfloat16 unaligned so I think as is is fine.

Comment (bertmaher) [reply torch/_inductor/fx_passes/pad_mm.py:191]: Cool, makes sense to me!

**Pull request closed without merging**